test/macros.jl

```diff
@@ -59,3 +59,15 @@ end
     @test_throws ErrorException g"\\"
 
 end
+
+@testset "@install GapObj" begin
+    a = GAP.evalstr("(1,2)")
+
+    struct TestType1 X::GapObj end
+    GAP.@install GapObj(x::TestType1) = x.X
+    @test GapObj(TestType1(a)) === a
+
+    struct TestType2 X::GapObj end
+    GAP.@install function GapObj(x::TestType2) return x.X; end
+    @test GapObj(TestType2(a)) === a
```

Comment:
Maybe also add a few tests for the error cases? I.e. trying to use the macro on functions other than `GapObj`, or on methods which have more or fewer arguments than expected

Reply:
When we call the macro in error cases then we run into `LoadError` before `@test_throws` can do something.
The usual solution seems to put the code of the macro into a helper function, and to test this helper function in the error cases.

Reply:
One can use `@macroexpand` for that purpose, e.g. in AbstractAlgebra we use this in a test:
```julia
        @test_throws ArgumentError @macroexpand @attr foo() = 1
```